video/out/vo_xv.c

```diff
@@ -561,18 +576,22 @@ static bool allocate_xvimage(struct vo *vo, int foo)
         if (!ctx->xvimage[foo])
             return false;
 
-        ctx->Shminfo[foo].shmid = shmget(IPC_PRIVATE,
-                                         ctx->xvimage[foo]->data_size,
-                                         IPC_CREAT | 0777);
-        ctx->Shminfo[foo].shmaddr = shmat(ctx->Shminfo[foo].shmid, 0, 0);
+        memcpy(ctx->shmname, "/tmp/mpv-xv-XXXXXXXXXX",
+               sizeof(ctx->shmname));
```

Comment:
this is an OOB-read because the string is surely smaller than the destination buffer

Reply:
most secure OpenBSD code